src/ffmpeg.js

```diff
@@ -94,56 +94,96 @@ class FFMPEG extends events.EventEmitter {
+            } else if (sub.codec === 'pgs') { // If program has PGS subs
+                tmpargs[vidStream] = '[v]'
+                if (typeof sub.index === 'undefined') { // If external subs
+                    console.log("PGS SUBS (external) - Not implemented..")
+                    tmpargs.splice(vidStream - 1, 0, "-filter_complex", `${iconOverlay}[v1];[v1]${deinterlace}[v]`)
+                } else {                    // Otherwise, internal/embeded pgs subs
+                    console.log("PGS SUBS (embeded)")
+                    tmpargs.splice(vidStream - 1, 0, "-filter_complex", `${iconOverlay}[v2];[v2]${deinterlace}[v1];[v1][0:${sub.index}]overlay[v]`)
+                }
+            } else if (sub.codec === 'srt' || sub.codec === 'ass') {
+                tmpargs[vidStream] = '[v]'
+                if (typeof sub.index === 'undefined') {
+                    console.log("SRT SUBS (external)")
+                    await this.createSubsFromStream(sub.key, lineupItem.start / 1000, lineupItem.duration / 1000, 0, `${process.env.DATABASE}/${uniqSubFileName}.ass`)
+                    tmpargs.splice(vidStream - 1, 0, "-filter_complex", `${iconOverlay}[v1];[v1]${deinterlace},ass=${process.env.DATABASE}/${uniqSubFileName}.ass[v]`)
+                } else {
+                    console.log("SRT SUBS (embeded) - This may take a few seconds..")
+                    await this.createSubsFromStream(lineupItem.file, lineupItem.start / 1000, lineupItem.duration / 1000, sub.index, `${process.env.DATABASE}/${uniqSubFileName}.ass`)
+                    tmpargs.splice(vidStream - 1, 0, "-filter_complex", `${iconOverlay}[v1];[v1]${deinterlace},ass=${process.env.DATABASE}/${uniqSubFileName}.ass[v]`)
+                }
+            } else { // Can't do VobSub's as plex only hosts the .idx file, there is no access to the .sub file.. Who the fuck uses VobSubs anyways.. SRT/ASS FTW
+                tmpargs[vidStream] = '[v]'
+                tmpargs.splice(vidStream - 1, 0, "-filter_complex", `${iconOverlay}[v1];[v1]${deinterlace}[v]`)
+                console.log("No Compatible Subtitles")
             }
-        } else { // Can't do VobSub's as plex only hosts the .idx file, there is no access to the .sub file.. Who the fuck uses VobSubs anyways.. SRT/ASS FTW
-            tmpargs[vidStream] = '[v]'
-            tmpargs.splice(vidStream - 1, 0, "-filter_complex", `${iconOverlay}[v1];[v1]${deinterlace}[v]`)
-            console.log("No Compatible Subtitles")
+
+            if (this.channel.icon !== '' && this.channel.overlayIcon && lineupItem.type === 'program') // Add the channel icon to ffmpeg input if enabled
+                tmpargs.splice(vidStream - 1, 0, '-i', this.channel.icon)
         }
 
-        if (this.channel.icon !== '' && this.channel.overlayIcon && lineupItem.type === 'program') // Add the channel icon to ffmpeg input if enabled
-            tmpargs.splice(vidStream - 1, 0, '-i', this.channel.icon)
```

Comment:
just wrapped this section in an if statement. If you direct stream video subtitles/channel icons are disabled